core/spec/models/spree/order_spec.rb

```diff
@@ -2161,43 +2161,79 @@ def validate(line_item)
 
     %w[customer_metadata admin_metadata].each do |metadata_type|
       describe metadata_type.to_s do
-        it "does not allow more than 6 keys" do
-          subject.send("#{metadata_type}=", invalid_metadata_keys)
+        context "when metadata validation is enabled" do
```

Comment:
this commit is great, because it introduces the change together with a test and a descriptive commit message 👍🏻 it would be great if the previous commits also would be like this.